pkgs/development/tools/detekt/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetchFromGitHub, unzip, makeWrapper, jre_headless }:
+
+stdenv.mkDerivation rec {
+  pname = "detekt";
+  version = "1.22.0";
+
+  src = fetchFromGitHub {
+    owner = "detekt";
+    repo = "detekt";
+    rev = "v${version}";
+    sha256 = "08jbnn3q3c7vp6jmj3gz67d5v9jlgd9f3dr706cbk8q2v9719gqh";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  preBuild = ''
+    export JAVA_HOME=${jre_headless}
+  '';
+
+  installPhase = ''
+    ./gradlew :detekt-cli:distZip
+    mkdir $out
+    unzip detekt-cli/build/distributions/detekt-cli-1.22.0.zip -d $out
+    mv $out/detekt-cli-${version}/* $out
```

Comment:
Why did you unzip into out only to be forced to clean it? Unzip in the local directory and simply move/rename the new unzip directory to $out as you did in this step.